### PR TITLE
Add simple test to show usage for Capnp

### PR DESCRIPTION
**Pull request by mishazharov**
This is a small PR to demonstrate how Capnp works (and that we're linking with it properly etc.)

### CODE REVIEW

File: src/Protos/operation.capnp [hunk 1]
```diff
@@ -1,5 +1,7 @@
 @0xf54292a58fc34e9a;
 
+using Cxx = import "/capnp/c++.capnp";
+$Cxx.namespace("caffeine_distributed::protos");
```
Comment (mishazharov) [inline src/Protos/operation.capnp:4]: This places the operation file into the `caffeine_distributed::protos` namespace

**Review by Phantomical — APPROVED** [review]
Generally LGTM! 

As a suggestion: maybe designate a folder within the binary directory for generated artifacts. That way you don't have to worry too much about adding all sorts of weird directories in the include path. (Adding  `${CMAKE_BINARY_DIR}/test/unit` to the include path seems weird to me)

File: src/Protos/operation.capnp (hunk 1 above)
Comment (Phantomical) [reply src/Protos/operation.capnp:4]: Nit: Can we just do `caffeine::protos`? I would rather everything be in the `caffeine` root namespace.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -43,15 +43,15 @@ file(
...
+set(CAPNPC_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/gen/caffeine/")
 file(MAKE_DIRECTORY ${CAPNPC_OUTPUT_DIR})
 
 capnp_generate_cpp(CAPNP_SRCS CAPNP_HDRS "${capnp_schemas}")
 list(APPEND sources ${CAPNP_SRCS})
 
 add_library(caffeine ${sources} ${headers})
 
-target_include_directories(caffeine PRIVATE "${CMAKE_CURRENT_BINARY_DIR}")
+target_include_directories(caffeine PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/gen/")
```
Comment (Phantomical) [inline src/CMakeLists.txt:54]: I think you want `${CMAKE_BINARY_DIR}/gen` here?

File: test/unit/CMakeLists.txt [hunk 3]
```diff
@@ -16,11 +16,12 @@ target_link_libraries(caffeine-unittest PRIVATE GTest::GTest)
 target_include_directories(caffeine-unittest PRIVATE "${CMAKE_SOURCE_DIR}/include")
 target_include_directories(caffeine-unittest PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
 target_include_directories(caffeine-unittest PRIVATE "${CMAKE_SOURCE_DIR}")
+target_include_directories(caffeine-unittest PRIVATE "${CMAKE_CURRENT_BINARY_DIR}")
```
Comment (Phantomical) [inline test/unit/CMakeLists.txt:19]: I think the same applies here? Not sure what's supposed to be at that path.

**Pull request merged**